olp-cpp-sdk-core/src/http/NetworkUtils.cpp

```diff
@@ -50,11 +50,13 @@ bool NetworkUtils::CaseInsensitiveStartsWith(const std::string& str1,
 size_t NetworkUtils::CaseInsensitiveFind(const std::string& str1,
                                          const std::string& str2,
                                          size_t offset /* = 0*/) {
-  while (offset < str1.length() - str2.length()) {
-    if (SimpleToUpper(str1[offset]) == SimpleToUpper(str2[0])) {
-      if (CaseInsensitiveStartsWith(str1, str2, offset)) return offset;
+  if (str2.length() && str1.length() >= str2.length()) {
```

Comment:
```suggestion
  if (!str2.empty() && str1.length() >= str2.length()) {
```
Prefer usage of `std::string::empty` when checking whether string is empty